src/main/java/com/mongodb/hibernate/translate/MqlTranslator.java

```diff
@@ -0,0 +1,595 @@
+import org.hibernate.sql.ast.tree.update.Assignment;
+import org.hibernate.sql.ast.tree.update.UpdateStatement;
+import org.hibernate.sql.exec.internal.JdbcParametersImpl;
+import org.hibernate.sql.exec.spi.JdbcOperation;
+import org.hibernate.sql.exec.spi.JdbcParameterBinder;
+import org.hibernate.sql.exec.spi.JdbcParameterBindings;
+import org.hibernate.sql.model.MutationOperation;
+import org.hibernate.sql.model.ast.ColumnWriteFragment;
+import org.hibernate.sql.model.ast.TableInsert;
+import org.hibernate.sql.model.ast.TableMutation;
+import org.hibernate.sql.model.internal.OptionalTableUpdate;
+import org.hibernate.sql.model.internal.TableDeleteCustomSql;
+import org.hibernate.sql.model.internal.TableDeleteStandard;
+import org.hibernate.sql.model.internal.TableInsertCustomSql;
+import org.hibernate.sql.model.internal.TableInsertStandard;
+import org.hibernate.sql.model.internal.TableUpdateCustomSql;
+import org.hibernate.sql.model.internal.TableUpdateStandard;
+import org.hibernate.sql.model.jdbc.JdbcMutationOperation;
+import org.mockito.Mockito;
+
+final class MqlTranslator<T extends JdbcOperation & MutationOperation> implements SqlAstTranslator<T> {
+
+    private final SessionFactoryImplementor sessionFactory;
+    private final Statement statement;
+
+    private final AstVisitorValueHolder astReturnValueHolder = AstVisitorValueHolder.emptyHolder();
+
+    private final List<JdbcParameterBinder> parameterBinders = new ArrayList<>();
+    private final JdbcParametersImpl jdbcParameters = new JdbcParametersImpl();
+
+    MqlTranslator(SessionFactoryImplementor sessionFactory, Statement statement) {
+        this.sessionFactory = sessionFactory;
+        this.statement = statement;
+    }
+
+    @Override
+    public SessionFactoryImplementor getSessionFactory() {
+        return sessionFactory;
+    }
+
+    @Override
+    public void render(SqlAstNode sqlAstNode, SqlAstNodeRenderingMode sqlAstNodeRenderingMode) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean supportsFilterClause() {
+        return true;
+    }
+
+    @Override
+    public QueryPart getCurrentQueryPart() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Stack<Clause> getCurrentClauseStack() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Set<String> getAffectedTableNames() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    @SuppressWarnings({"unchecked"})
+    public T translate(JdbcParameterBindings jdbcParameterBindings, QueryOptions queryOptions) {
+        if (statement instanceof TableMutation) {
+            TableMutation<T> tableMutation = (TableMutation<T>) statement;
+            if (tableMutation instanceof TableInsert) {
+                return translateTableMutation(tableMutation);
+            } else {
+                return (T) Mockito.mock(JdbcMutationOperation.class);
```

Comment:
This is strange.  I've never seen Mockito used outside of test code.  Can you explain why this is here, and what the alternative is if we don't want this dependency?